kernel/src/engine/arrow_expression/mod.rs

```diff
@@ -86,7 +86,26 @@ impl Scalar {
             Byte(val) => append_val_as!(array::Int8Builder, *val),
             Float(val) => append_val_as!(array::Float32Builder, *val),
             Double(val) => append_val_as!(array::Float64Builder, *val),
-            String(val) => append_val_as!(array::StringBuilder, val),
+            String(val) => {
```

Comment:
if we know that the only place that calls into this is above in `to_array` (which it is afaict), and `to_array` figures the type out by doing `let data_type = ArrowDataType::try_from_kernel(&self.data_type())?;` then we _know_ that the builder is always of a large type if that's what our schema conversion does, so we should never need to handle the small case?

Alternately, if we want to have an option to do both, let's check what type we converted into in `to_array` and pass an extra `is_large` argument or something that let's us do this without having to just try and see what works.